aws/adapter.go

```diff
@@ -1009,3 +1048,53 @@ func nonTargetedASGs(ownedASGs, targetedASGs map[string]*autoScalingGroupDetails
 
 	return nonTargetedASGs
 }
+
+// SetTargetsOnCNITargetGroups implements desired state for CNI target groups
+// by polling the current list of targets thus creating a diff of what needs to be added and removed.
+func (a *Adapter) SetTargetsOnCNITargetGroups(endpoints []string) error {
+	if a.TargetCNI.TargetGroupARNs == nil || len(a.TargetCNI.TargetGroupARNs) == 0 {
+		return ErrNoTargetGroupARNs
+	}
+	for _, targetGroupARN := range a.TargetCNI.TargetGroupARNs {
+		tgh, err := a.elbv2.DescribeTargetHealth(&elbv2.DescribeTargetHealthInput{TargetGroupArn: &targetGroupARN})
+		if err != nil {
+			return fmt.Errorf("unable to describe target health %w", err)
+		}
+		registeredInstances := make([]string, len(tgh.TargetHealthDescriptions))
+		for i, target := range tgh.TargetHealthDescriptions {
+			registeredInstances[i] = *target.Target.Id
+		}
+		toregister := difference(endpoints, registeredInstances)
+		if len(toregister) > 0 {
+			log.Info("Registering CNI targets: ", toregister)
+			err := registerTargetsOnTargetGroups(a.elbv2, a.TargetCNI.TargetGroupARNs, toregister)
```

Comment:
This feels wrong: it registers all of `toregister` on all `a.TargetCNI.TargetGroupARNs`.
Maybe it should calculate all to register and all to de-register and call `registerTargetsOnTargetGroups/deregisterTargetsOnTargetGroups` outside of the loop. 
Or call `registerTargetsOnTargetGroups/deregisterTargetsOnTargetGroups` only for the current iteration target group.